src/briefcase/commands/base.py

```diff
@@ -525,10 +571,10 @@ def download_url(self, url, download_path):
                         downloaded += len(data)
                         f.write(data)
                         done = int(50 * downloaded / total)
-                        print('\r{}{} {}%'.format('#' * done, '.' * (50-done), 2*done), end='', flush=True)
-            print()
+                        self.input.print('\r{}{} {}%'.format('#' * done, '.' * (50-done), 2*done), end='', flush=True)
```

Comment:
Not sure why this is going through `input`; it's ostensibly a progress bar, so it's 100% output functionality; and the lone "print" on the next line is what does the newline flush.

I'd be more inlined to wrap this as a literal "show progress" function - possibly a context manager where exiting does the implied newline at the end?

Reply:
I implemented a progress bar context manager in Console.